extensions/sql-database-projects/src/models/project.ts

```diff
@@ -508,18 +508,27 @@ export class Project implements ISqlProject {
 	}
 
 	public getProjectTargetVersion(): string {
-		// check for invalid DSP
-		if (this.projFileXmlDoc.getElementsByTagName(constants.DSP).length !== 1 || this.projFileXmlDoc.getElementsByTagName(constants.DSP)[0].childNodes.length !== 1) {
-			throw new Error(constants.invalidDataSchemaProvider);
+		let dsp: string | undefined;
+
+		try {
+			dsp = this.evaluateProjectPropertyValue(constants.DSP);
+		}
+		catch {
+			// We will throw specialized error instead
 		}
 
-		let dsp: string = this.projFileXmlDoc.getElementsByTagName(constants.DSP)[0].childNodes[0].data;
+		// Check if DSP is missing or invalid
+		if (!dsp) {
+			throw new Error(constants.invalidDataSchemaProvider);
+		}
 
 		// get version from dsp, which is a string like Microsoft.Data.Tools.Schema.Sql.Sql130DatabaseSchemaProvider
-		// remove part before the number
-		let version: any = dsp.substring(constants.MicrosoftDatatoolsSchemaSqlSql.length);
-		// remove DatabaseSchemaProvider
-		version = version.substring(0, version.length - constants.databaseSchemaProvider.length);
+		// Remove prefix and suffix to only get the actual version number/name. For the example above the result
+		// should be just '130'.
+		const version =
```

Comment:
I was not really sure what is going on here and why `substring` is called twice. I just made it start from where there first `substring` call was cutting it, instead of 0. Let me know, if I'm missing something.

Reply:
it was just getting called twice for clarity on what was getting removed. Calling it once like this is fine